extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,236 @@
+						crs,
+						0),
+				clipRange(
+						coord.y,
+						crs,
+						1),
+				clipRange(
+						coord.z,
+						crs,
+						2));
+	}
+
+	/**
+	 * This is perhaps a brain dead approach to do this, but it does handle wrap
+	 * around cases
+	 * 
+	 * @param val
+	 * @param bound
+	 * @return
+	 */
+	private static double clipRange(
+			final double val,
+			final CoordinateReferenceSystem crs,
+			int dimension ) {
+		final CoordinateSystem coordinateSystem = crs.getCoordinateSystem();
+		if (coordinateSystem.getDimension() > dimension) {
+			double lowerBound = coordinateSystem.getAxis(
+					dimension).getMinimumValue();
+			double bound = coordinateSystem.getAxis(
+					dimension).getMaximumValue() - lowerBound;
+			double sign = sign(val);
+			// re-scale to 0 to n, then determine how many times to 'loop
+			// around'
+			double mult = Math.floor(Math.abs((val + sign * (-1.0 * lowerBound)) / bound));
+			return val + mult * bound * sign * (-1.0);
+		}
+		return val;
+	}
+
+	/**
+	 * Convert meters to decimal degrees based on widest point
+	 * 
+	 * @throws TransformException
+	 */
+	private static double distanceToDegrees(
+			final CoordinateReferenceSystem crs,
+			final Geometry geometry,
+			final double meters )
+			throws TransformException {
+		final GeometryFactory factory = geometry.getFactory();
+		return (geometry instanceof Point) ? geometry.distance(farthestPoint(
+				crs,
+				(Point) geometry,
+				meters)) : distanceToDegrees(
+				crs,
+				geometry.getEnvelopeInternal(),
+				factory == null ? new GeometryFactory() : factory,
+				meters);
+	}
+
+	private static double distanceToDegrees(
+			final CoordinateReferenceSystem crs,
+			final Envelope env,
+			final GeometryFactory factory,
+			final double meters )
+			throws TransformException {
+		return Collections.max(Arrays.asList(
+				distanceToDegrees(
+						crs,
+						factory.createPoint(new Coordinate(
+								env.getMaxX(),
+								env.getMaxY())),
+						meters),
+				distanceToDegrees(
+						crs,
+						factory.createPoint(new Coordinate(
+								env.getMaxX(),
+								env.getMinY())),
+						meters),
+				distanceToDegrees(
+						crs,
+						factory.createPoint(new Coordinate(
+								env.getMinX(),
+								env.getMinY())),
+						meters),
+				distanceToDegrees(
+						crs,
+						factory.createPoint(new Coordinate(
+								env.getMinX(),
+								env.getMaxY())),
+						meters)));
+	}
+
+	/** farther point in longitudinal axis given a latitude */
+
+	private static Point farthestPoint(
+			final CoordinateReferenceSystem crs,
+			final Point point,
+			final double meters ) {
+		final GeodeticCalculator calc = new GeodeticCalculator(
+				crs);
+		calc.setStartingGeographicPoint(
+				point.getX(),
+				point.getY());
+		calc.setDirection(
+				90,
+				meters);
+		Point2D dest2D = calc.getDestinationGeographicPoint();
+		// if this flips over the date line then try the other direction
+		if (sign(dest2D.getX()) != sign(point.getX())) {
```

Comment:
I think this really should cover us as much as we can for the dateline.  In the last commit, this check was really all I was alluding to.